txnsync/outgoing_test.go

```diff
@@ -0,0 +1,294 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package txnsync
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/go-algorand/util/bloom"
+	"github.com/algorand/go-algorand/util/timers"
+)
+
+type mockAsyncLogger struct {
+	logging.Logger
+}
+
+func (m mockAsyncLogger) outgoingMessage(mstat msgStats) {
+}
+
+func (m mockAsyncLogger) incomingMessage(mstat msgStats) {
+}
+
+func (m mockAsyncLogger) Infof(string, ...interface{}) {}
+
+func TestAsyncMessageSent(t *testing.T) {
+	a := require.New(t)
+
+	var s syncState
+	s.clock = timers.MakeMonotonicClock(time.Now())
+	s.log = mockAsyncLogger{}
+
+	asyncEncoder := messageAsyncEncoder{
+		state: &s,
+		messageData: sentMessageMetadata{
+			encodedMessageSize:  0,
```

Comment:
You don't need to initialize to the default values.
If you only initialize the values you care about, and which are different from the default values, the code will be clearer.